internal/integratedservices/registry.go

```diff
@@ -74,6 +96,105 @@ func (r integratedServiceOperatorRegistry) GetIntegratedServiceOperator(integrat
 	return nil, errors.WithStack(UnknownIntegratedServiceError{IntegratedServiceName: integratedServiceName})
 }
 
+func (r integratedServiceOperatorRegistry) DisableServiceInstance(ctx context.Context, clusterID uint) error {
+	clusterClient, err := r.k8sClientForCluster(ctx, clusterID)
+	if err != nil {
+		return errors.Wrap(err, "failed to build cluster client")
+	}
+
+	lookupISvcs := &v1alpha1.ServiceInstanceList{}
+	if err := clusterClient.List(ctx, lookupISvcs); err != nil {
+		if meta.IsNoMatchError(err) {
+			return nil
+		}
+		return errors.Wrap(err, "failed to retrieve service instance list")
+	}
+
+	kubeConfig, err := r.kubeConfigFn.GetKubeConfig(ctx, clusterID)
+	if err != nil {
+		return errors.WrapIf(err, "failed to get K8S config")
+	}
+
+	restCfg, err := clientcmd.RESTConfigFromKubeConfig(kubeConfig)
+	if err != nil {
+		return errors.Wrap(err, "failed to create rest config from cluster configuration")
+	}
+
+	cli, err := client.New(restCfg, client.Options{Scheme: r.scheme})
+	if err != nil {
+		return errors.Wrap(err, "failed to create the client from rest configuration")
+	}
+
+	for _, item := range lookupISvcs.Items {
+		item.Spec.Enabled = utils.BoolPointer(false)
```

Comment:
check if the spec is already disabled, because then we don't have to disable it again